app/org/sagebionetworks/bridge/hibernate/HibernateAccountDao.java

```diff
@@ -172,51 +179,59 @@ public Account reauthenticate(Study study, SignIn signIn) {
         if (!study.isReauthenticationEnabled()) {
             throw new UnauthorizedException("Reauthentication is not enabled for study: " + study.getName());    
         }
-        Account hibernateAccount = fetchHibernateAccount(signIn);
-        return authenticateInternal(study, hibernateAccount, hibernateAccount.getReauthTokenAlgorithm(),
-                hibernateAccount.getReauthTokenHash(), signIn.getReauthToken(), "reauth token", signIn);
+        Account account = fetchHibernateAccount(signIn);
+        if (account.getReauthTokenHash() != null) {
+            // Rotate the token to the secrets table before verifying. This migrates the tables if
+            // a reauthentication call occurs before we run a migration on this record.
+            AccountSecret secret = AccountSecret.create();
+            secret.setAccountId(account.getId());
+            secret.setAlgorithm(account.getReauthTokenAlgorithm());
+            secret.setHash(account.getReauthTokenHash());
+            secret.setType(AccountSecretType.REAUTH);
+            secret.setCreatedOn(account.getReauthTokenModifiedOn());
+            hibernateHelper.create(secret, null);
```

Comment:
We can't because createSecret would re-hash the already hashed token. Because this is just for migration, I didn't want to add a special method in the DAO for this, but if you want I can add. I intend to migrate and then remove this code.